cartoframes/io/carto.py

```diff
@@ -93,6 +96,11 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
         ValueError: if the dataframe or table name provided are wrong or the if_exists param is not valid.
 
     """
+    def estimate_csv_size(gdf):
+        n = min(100, len(gdf))
+        return len(''.join([x.decode("utf-8") for x in
+                            _compute_copy_data(gdf.sample(n=n), get_dataframe_columns_info(gdf))])) * len(gdf) / n
```

Comment:
I would extract this `get_dataframe_columns_info(gdf)` to a variable columns_info before the loop.

Also, we could try something like `sum([len(x) for x in ...])` to avoid the decode and join so it will be a bit faster.

Reply:
nice catch, on it!